include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -625,24 +625,29 @@ struct __early_exit_find_or
             using _ShiftedIdxType =
                 typename ::std::conditional<_OrTagType::value, decltype(__init_index + __i * __shift),
                                             decltype(__found_local.load())>::type;
-            _IterSize __current_iter = oneapi::dpl::__internal::__invoke_if_else(
-                _BackwardTagType{}, [__n_iter, __i]() { return __n_iter - 1 - __i; }, [__i]() { return __i; });
+
+            _IterSize __current_iter =  [__n_iter, __i](){
```

Comment:
It seems to me that we may rewrite this without lambdas.

Reply:
I used an [immediately invoked function expression](https://en.wikipedia.org/wiki/Immediately_invoked_function_expression) here to avoid having to initialize `__current_iter` with a default value and only then conditionally assigning it. But I did some testing on godbolt (https://godbolt.org/z/e67qancP4) and it seems that the code generated is the same with and without the lambda. So I will rewrite this to remove the lambda , which should make it clearer.